Madness/SourcePos.swift

```diff
@@ -0,0 +1,49 @@
+//  Copyright (c) 2014 Josh Vera. All rights reserved.
+
+import Prelude
+
+public typealias Line = Int
+public typealias Column = Int
+
+public struct SourcePos<Index: ForwardIndexType>: Equatable {
```

Comment:
Capturing this instead of just the index avoids us having to do O(n) traversals every time we want to correlate a string index with line/column numbers, right?